src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java

```diff
@@ -463,11 +468,66 @@ public Object visit(ASTCollectionLiteral node, Object data) {
       node.setIrElement(range);
       return data;
     }
+    if (node.isComprehension()) {
+      Block comprehension = createCollectionComprehension(context, node.getType(), expressions);
+      comprehension.setASTNode(node);
+      context.objectStack.push(comprehension);
+      return data;
+    }
     CollectionLiteral.Type type = CollectionLiteral.Type.valueOf(node.getType());
     context.objectStack.push(new CollectionLiteral(type, expressions));
     return data;
   }
 
+  private Block createCollectionComprehension(Context context, String type, List<ExpressionStatement> expressions) {
+    // TODO: special case for iter that create a generator
```

Comment:
yes, this feature will be added in a separate PR dealing more specifically with generators and lazyness. Do you rather I remove the TODO comment (or make it a NOTE)?